### PR TITLE
after-the-deadline: Avoid accessing the superglobals directly

### LINKED ISSUE #4308: Avoid accessing the superglobals directly
With CodeClimate now setup, I went thru some of the items reported. I suggest going thru the security issues first.

https://codeclimate.com/github/Automattic/jetpack/issues?category=security

For starters it seems instead of accessing Superglobals directly we should be using `filter_input` or something similar. This comes up quite a bit.

http://php.net/manual/en/function.filter-input.php

### PR DESCRIPTION
**Pull request by shantanu2704**
See #4308

This is the first pull request in a series of pull requests to fix this issue.
Since there would be too many changes to review in one PR, I'm sending in one PR per module of Jetpack

Changes proposed in this Pull Request:
All occurrences of $_SERVER, $_GET and $_POST have been replaced with filter_input (or filter_has_var wherever necessary) in the afrer-the-deadline module.

PRs for other modules to follow.

### CODE REVIEW

**Comment by oskosk** [comment]
Hi @shantanu2704 . Thanks for this. Can you add testing instructions in the PR description with a series of steps that allows us to check that current behaviour affected by the changes remains the same after your changes ?

File: modules/after-the-deadline/config-options.php [hunk 1]
```diff
@@ -117,8 +117,10 @@ function AtD_get_options( $user_id, $name ) {
  */
 function AtD_update_options( $user_id, $name ) {
 	/* We should probably run $_POST[name] through an esc_*() function... */
-	if ( isset( $_POST[$name] ) && is_array( $_POST[$name] ) ) {
-		$copy = array_map( 'strip_tags', array_keys( $_POST[$name] ) );
+	$post_name = filter_input(INPUT_POST, $name, FILTER_DEFAULT , FILTER_REQUIRE_ARRAY);
```
Comment (eliorivero) [inline modules/after-the-deadline/config-options.php]: This needs to fix the spacing.

**Comment by zinigor** [comment]
As much as I love native implementations, I have to agree with @eliorivero on this. Judging by the documentation and comments, `filter_input` works like a black box with some weird quirks, hiding its logic. Whereas now everything is plain, simple and self-explanatory.

Sorry, but I don't see any benefit either. Perhaps you can show us an example where `filter_input` may be better than doing it old school?

File: modules/after-the-deadline/config-unignore.php [hunk 2]
```diff
@@ -15,7 +15,7 @@ function AtD_ignore_call() {
 	check_admin_referer( 'atd_ignore' );
 
 	$ignores = explode( ',', AtD_get_setting( $user->ID, 'AtD_ignored_phrases') );
-	array_push( $ignores, $_GET['phrase'] );
+	array_push( $ignores, filter_input( INPUT_GET, 'phrase' ) );
```
[the remark this answers is not in the document]
Comment (zinigor) [reply modules/after-the-deadline/config-unignore.php]: There is a slight difference. If no `phrase` is passed, the former would result in a notice, while the latter would just return NULL. I'm not sure either is a great way to treat input, though.

**Comment by keoshi** [comment]
@shantanu2704 Hi again! Wondering if you saw the comments and feedback on this PR. Would you mind addressing them, please? Thank you once again, and sorry for the barrage of notifications!

**Comment by shantanu2704** [comment]
I am sorry I don't think I have the bandwidth right now. Closing the PR. Sorry for the trouble!

**Comment by keoshi** [comment]
No worries, and thank you for your contributions, @shantanu2704 !

**Pull request closed without merging**